internal/logging/config.go

```diff
@@ -0,0 +1,110 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	// FileOutput is an output option for logging to a file.
+	FileOutput string = "file"
+
+	// StdOutput is an output option for logging to stdout.
+	StdOutput string = "stdout"
+)
+
+// LoggerConfig is the configuration of a logger.
+type LoggerConfig struct {
+	Output string             `yaml:"output"`
+	Level  zapcore.Level      `yaml:"level"`
+	File   *lumberjack.Logger `yaml:"file"`
+}
+
+// NewLoggerConfig returns a logger config. If configPath is not
+// set, stdout logging will be enabled.
+func NewLoggerConfig(configPath string) (*LoggerConfig, error) {
+	conf := &LoggerConfig{
+		Output: StdOutput,
+		Level:  zapcore.InfoLevel,
+	}
+
+	if configPath == "" {
```

Comment:
Actually, I think this is because of filepath.Clean, which I introduced after this due to gosec. So I think you're correct, but now we do need it because of the clean.